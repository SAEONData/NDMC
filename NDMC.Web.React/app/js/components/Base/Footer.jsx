'use strict'
/**
 * @ignore
 * Imports
 */
import React from 'react'
import { Col, Row, Modal, ModalBody, ModalHeader } from 'mdbreact';
import { DEAGreen } from '../../config/colours.cfg'
import { footerContent } from '../../../data/footerConfig'

/**
 * The Footer class for displaying a footer for references and additional information
 * @class
 */
class Footer extends React.Component {
  constructor(props) {
    super(props);
    this.toggleModal = this.toggleModal.bind(this)
    this.renderLinks = this.renderLinks.bind(this)
    this.handleResize = this.handleResize.bind(this)
    this.state = {
      showModal: false,
      modalHeader: "",
      modalSrc: "",
      isNarrowDisplay: false
    }
  }

  componentDidMount() {
    window.addEventListener("resize", this.handleResize);
  }

  componentWillUnmount() {
    window.removeEventListener("resize", this.handleResize)
  }

  /**
   * Handle page resizing for narrow display
   */
  handleResize() {
    this.setState({
      isNarrowDisplay: (window.innerWidth < 900)
    })
  }

  /**
   * Handle toggling header/footer model
   * @param {string} state The state to set 
   * @param {string} header The custom defined header
   * @param {string} src Data sources
   */
  toggleModal(state, header = "", src = "") {
    this.setState({
      showModal: state,
      modalHeader: header,
      modalSrc: src
    })
  }

  /**
   * Handle what data to render
   * @param {*} data 
   */
  renderSections(data) {
    let sections = []
    for (let i = 0; i < 4; i++) {
      let section = data.sections[i]
      sections.push(
        <Col key={`section_${i + 1}`} md="3">
          <h4 style={{ marginBottom: "15px" }}><b>{section.text}</b></h4>
          {this.renderLinks(section)}
        </Col>
      )
    }
    return sections
  }

  /**
   * Handle what links to render on footer
   * @param {object} section The render section object
   */
  renderLinks(section) {
    let links = []
    for (let i = 0; i < section.links.length; i++) {
      let link = section.links[i]
      if (link.text) {
        links.push(
          <div
            key={`link_${i + 1}`}
            style={{
              cursor: link.link ? "pointer" : "default",
              fontWeight: link.link ? "400" : "regular"
            }}
            onClick={() => {
              if (link.link) {
                this.toggleModal(true, link.text, link.link)
              }
            }}>
            {link.text}
          </div>
        )
      }
      else if (link.src) {
        links.push(
          <img
            key={`link_${i + 1}`}
            src={link.src}
            style={{
              width: link.width,
              cursor: link.link ? "pointer" : "default"
            }} />
        )
      }
    }
    return links
  }

  render() {
    let { showModal, modalHeader, modalSrc } = this.state
    return (
      <>
        <div style={{
          padding: "5px 45px 5px 45px",
          borderTop: "1px solid gainsboro",
          borderLeft: "1px solid gainsboro",
          borderRight: "1px solid gainsboro",
          backgroundColor: "white",
          color: "black"
        }}>
          <br />
          <Row>
            {this.renderSections(footerContent)}
          </Row>
          <br />
        </div>
        <div style={{
          padding: "10px 45px 10px 45px",
          backgroundColor: DEAGreen,
          color: "white"
        }}>
          <Row>
            <Col md="12">
              <b>
                Created by
                <span> <a style={{ color: "white" }} href="http://www.saeon.ac.za" target="saeon"><strong>SAEON</strong></a> </span>
                using
                <span> <a style={{ color: "white" }} href="https://reactjs.org/" target="react"><strong>React</strong></a>, </span>
                <span> <a style={{ color: "white" }} href="https://mdbootstrap.com/react" target="mdb"><strong> MDBootstrap</strong></a>, </span>
                <span> <a style={{ color: "white" }} href="https://ant.design/" target="antdesign"><strong> Ant Design </strong></a> </span>
                and SAEON Open Data Platform APIs.
              </b>
              <br />
              Copyright &copy; {(new Date().getFullYear())}
            </Col>
          </Row>
        </div>
        <Modal isOpen={showModal} toggle={() => this.toggleModal(false)} size="fluid" style={{ width: "95%" }} >
          <ModalHeader toggle={() => this.toggleModal(false)}>{modalHeader}</ModalHeader>
          <ModalBody>
            <iframe
              style={{
                width: "100%",
                height: "500px",
                margin: "0px",
                border: "1px solid gainsboro",
                // backgroundImage: `url(${loader})`,
                // backgroundRepeat: "no-repeat",
                // backgroundPosition: "50% 50%"
              }}
              src={modalSrc}
            />
          </ModalBody>
        </Modal>
      </>
    )
  }
}
export default Footer