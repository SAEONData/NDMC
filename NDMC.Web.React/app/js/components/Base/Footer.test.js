import React from 'react'
import { shallow } from 'enzyme'

import Footer from './Footer'

describe('Footer', () => {
  it('should always render at bottom of page', () => {
    const footer = shallow(<Footer />)

    expect(footer).toMatchSnapshot()
  })
})
