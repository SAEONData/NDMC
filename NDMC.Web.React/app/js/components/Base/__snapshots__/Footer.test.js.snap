// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should always render at bottom of page 1`] = `
<Fragment>
  <div
    style={
      Object {
        "backgroundColor": "white",
        "borderLeft": "1px solid gainsboro",
        "borderRight": "1px solid gainsboro",
        "borderTop": "1px solid gainsboro",
        "color": "black",
        "padding": "5px 45px 5px 45px",
      }
    }
  >
    <br />
    <Row
      tag="div"
    >
      <Col
        key="section_1"
        lg={null}
        md="3"
        sm={null}
        tag="div"
        xl={null}
        xs={null}
      >
        <h4
          style={
            Object {
              "marginBottom": "15px",
            }
          }
        >
          <b>
            Technology
          </b>
        </h4>
        <div
          key="link_1"
          onClick={[Function]}
          style={
            Object {
              "cursor": "default",
              "fontWeight": "regular",
            }
          }
        >
          DEA NHE uses open source, government funded facilities provided by SAEON. The DST and NRF funds the SAEON Open Data Platform (ODP) and associated portals. Developed by SAEON on behalf of DST, DEA, DRDLR, and other stakeholders.
        </div>
      </Col>
      <Col
        key="section_2"
        lg={null}
        md="3"
        sm={null}
        tag="div"
        xl={null}
        xs={null}
      >
        <h4
          style={
            Object {
              "marginBottom": "15px",
            }
          }
        >
          <b>
            Legal
          </b>
        </h4>
        <div
          key="link_1"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
              "fontWeight": "400",
            }
          }
        >
          Disclaimer
        </div>
        <div
          key="link_2"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
              "fontWeight": "400",
            }
          }
        >
          Terms and Conditions
        </div>
        <div
          key="link_3"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
              "fontWeight": "400",
            }
          }
        >
          Data Licenses
        </div>
        <div
          key="link_4"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
              "fontWeight": "400",
            }
          }
        >
          Privacy
        </div>
      </Col>
      <Col
        key="section_3"
        lg={null}
        md="3"
        sm={null}
        tag="div"
        xl={null}
        xs={null}
      >
        <h4
          style={
            Object {
              "marginBottom": "15px",
            }
          }
        >
          <b>
            SAEON ODP
          </b>
        </h4>
        <div
          key="link_1"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
              "fontWeight": "400",
            }
          }
        >
          Open Data Platform
        </div>
        <div
          key="link_2"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
              "fontWeight": "400",
            }
          }
        >
          Contribute
        </div>
        <div
          key="link_3"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
              "fontWeight": "400",
            }
          }
        >
          Contact Us
        </div>
      </Col>
      <Col
        key="section_4"
        lg={null}
        md="3"
        sm={null}
        tag="div"
        xl={null}
        xs={null}
      >
        <h4
          style={
            Object {
              "marginBottom": "15px",
            }
          }
        >
          <b>
            Funding
          </b>
        </h4>
        <img
          key="link_1"
          src={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "cursor": "default",
              "width": "100%",
            }
          }
        />
      </Col>
    </Row>
    <br />
  </div>
  <div
    style={
      Object {
        "backgroundColor": "#8EAC85",
        "color": "white",
        "padding": "10px 45px 10px 45px",
      }
    }
  >
    <Row
      tag="div"
    >
      <Col
        lg={null}
        md="12"
        sm={null}
        tag="div"
        xl={null}
        xs={null}
      >
        <b>
          Created by
          <span>
             
            <a
              href="http://www.saeon.ac.za"
              style={
                Object {
                  "color": "white",
                }
              }
              target="saeon"
            >
              <strong>
                SAEON
              </strong>
            </a>
             
          </span>
          using
          <span>
             
            <a
              href="https://reactjs.org/"
              style={
                Object {
                  "color": "white",
                }
              }
              target="react"
            >
              <strong>
                React
              </strong>
            </a>
            , 
          </span>
          <span>
             
            <a
              href="https://mdbootstrap.com/react"
              style={
                Object {
                  "color": "white",
                }
              }
              target="mdb"
            >
              <strong>
                 MDBootstrap
              </strong>
            </a>
            , 
          </span>
          <span>
             
            <a
              href="https://ant.design/"
              style={
                Object {
                  "color": "white",
                }
              }
              target="antdesign"
            >
              <strong>
                 Ant Design 
              </strong>
            </a>
             
          </span>
          and SAEON Open Data Platform APIs.
        </b>
        <br />
        Copyright Â© 
        2019
      </Col>
    </Row>
  </div>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransitionTimeout={150}
    fade={true}
    isOpen={false}
    modalTransitionTimeout={300}
    role="dialog"
    size="fluid"
    style={
      Object {
        "width": "95%",
      }
    }
    tabIndex="-1"
    toggle={[Function]}
    zIndex={1050}
  >
    <ModalHeader
      closeAriaLabel="Close"
      tag="h4"
      toggle={[Function]}
    />
    <ModalBody>
      <iframe
        src=""
        style={
          Object {
            "border": "1px solid gainsboro",
            "height": "500px",
            "margin": "0px",
            "width": "100%",
          }
        }
      />
    </ModalBody>
  </Modal>
</Fragment>
`;
